accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,199 @@
+
+The main focus is to develop the integration between HUB server and peripherals through API only and re-using existing API methods as much as possible. However, some new custom methods will be needed.
+
+Configuration should be pushed from the HUB to peripherals, to avoid the need of direct management of the peripherals.
+
+For now this solution is focused on software channels only, but can be extended to synchronize more data types.
+
+
+## Define connection between HUB and Peripherals
+
+We can follow a similar approach to what exists on ISSv1. On the hub side we can define multiple peripheral servers to connect to by providing the FQDN and an authentication token.
+On the peripheral side we also need to define the Hub server FQDN and an Authentication token.
+
+Each peripheral server can only have 1 Hub server (master). This will avoid dealing with problems like channel label conflicts between multiple masters. Configure a hub server will block access to a set of menu items like: "Admin" -> "Setup wizard" -> "Organization Credentials" and "Products" and "PAYG Connection" (similar to what we already have for ISSv1).
+We cannot add Hub connection if SCC credentials are defined, they should be mutually excluded.
+
+We can re-use and improve existing ISSv1 database tables to save the needed data.
+
+
+## Hub as a proxy for SCC data
+
+The SUSE Manager server needs a set of metadata to be able to operate. Currently that metadata is provided by SCC directly or, in the case of PAYG, provided by the cloud RMT infrastructure. We should also provide this data in SUSE Manager HUB to be consumed by the peripheral servers.
+The minimal endpoints to be provided are:
+   - "/suma/product_tree.json"
+   - "/connect/organizations/products/unscoped"
+
+On top of this, we should also provide an endpoint for peripherals to send status data needed by SCC (example of this is the minions registered and hardware information). Peripheral servers should send this data to the HUB instead of SCC, and the HUB server should consolidate it and send it to SCC.
+
+## Peripheral software channels creation
+
+We need a mechanism to create the channels in the peripheral servers (vendor and CLM's) in the desired organization. The peripheral channel creation must be done from the HUB server through an API. Since we are making special channel creation (defined next), those API methods should be available to server-to-server communication only.
+
+For each peripheral we should define the organization mapping between HUB organizations and peripheral ones, following the rules (similiar to what we have on ISSv1):
+  - Each Hub organization can only be mapped to one peripheral organizations
+  - Peripheral organization can receive data from several different HUB organizations
```

Comment:
Why should we keep following this design? I mean what's the downside the of peripheral org to receive data from only one Hub org?

Reply:
I don't see a problem with one peripheral organization receiving data from one HUB organization. I was trying to keep it flexible.
This way, user can have Organizations on the HUB and decide to which peripheral organization they want to map it. If they select the same org to map on a peripheral, that would merge them in there. This is safe to do since channel labels are unique in the system, doesn't matter the organization.

Reply:
@admd I was now looking at how we can also synchronize configuration channels. For that we cannot have a peripheral receiving data from several different HUB organizations since the channel label is not unique across orgs. I will change this requirement to only allow one org.